site_scons/site_tools/extra/extra.py

```diff
@@ -47,7 +53,31 @@ def _supports_custom_format(clang_exe):
     if match and int(match.group(1)) >= MIN_FORMAT_VERSION:
         return True
 
-    print('Custom .clang-format wants version {}+. Using Mozilla style.'.format(MIN_FORMAT_VERSION))
+    print(f'Custom .clang-format wants version {MIN_FORMAT_VERSION}+. Using Mozilla style.')
+    return False
+
+
+def _supports_correct_style(clang_exe):
+    """Checks if the version of clang-format is 14.0.5 or newer which is required to correctly
+    reformat code for landing"""
+
+    try:
+        rawbytes = subprocess.check_output([clang_exe, "-version"])
+        output = rawbytes.decode('utf-8')
+    except subprocess.CalledProcessError:
+        return False
+
+    match = re.search(r'version ([\d+\.]+) ', output)
+    if match:
+        parts = match.group(1).split('.')
+        if int(parts[0]) < 14:
+            return False
+        if int(parts[1]) > 0:
+            return True
+        if int(parts[2]) < 5:
+            return False
+        return True
```

Comment:
FYI, there is a function for this
```
from packaging import version
version.parse('14.0.5') <= version.parse('14.0.4')
False
version.parse('14') <= version.parse('14.0.5')
True
```

Reply:
Awesome, will update.  This wasn't great code.

Reply:
Nice, but I'm not prepared to modify the rpm specfile for just this change.

Reply:
Sorry, I thought that module was built-in :(